governance/shutdown.go

```diff
@@ -118,8 +128,72 @@ func (w *GovernanceWorker) nodeShutdown(cmd *NodeShutdownCommand) {
 	glog.V(3).Infof(logString(fmt.Sprintf("node shutdown process complete.")))
 }
 
+// This function is called due to the node pattern change on the exchange while
+// the device is registered with a different pattern.
+// It will remove the agreements, workloads etc, cleanup the exchange node and
+// local device so that the node can be registered again.
+func (w *GovernanceWorker) nodeShutDownForPattenChanged(dev *persistence.ExchangeDevice, new_pattern string) {
+
+	glog.V(3).Infof(logString(fmt.Sprintf("begin node shutdown process for pattern change.")))
+
+	// Clear the Pattern and RegisteredServices array in the node’s exchange resource. We have to leave the
+	// public key so that the node can send messages to an agbot. Removing the pattern and RegisteredServices
+	// will prevent the exchange from finding the node and thereby prevent agobts from trying to make new agreements.
+	// It will keep the user input.
+	if err := w.clearNodePatternAndMS(true); err != nil {
+		w.continueWithError(logString(err.Error()))
+	}
+
+	// Cancel all agreements, all workload containers and networks will automatically terminate.
+	if err := w.terminateAllAgreements(producer.TERM_REASON_NODE_PATTERN_CHANGED); err != nil {
+		w.completedWithError(logString(err.Error()))
+		return
+	}
+
+	// Remove the node’s messaging public key from the node’s exchange resource and delete the node’s message key pair from the filesystem.
+	if err := w.patchNodeKey(); err != nil {
+		w.continueWithError(logString(err.Error()))
+	}
+
+	// Add the new pattern name back to the node for the pattern change case, It is safe to add it back because the public keys are cleared
+	// from the node
+	if err := w.patchNodePattern(new_pattern); err != nil {
+		w.completedWithError(logString(err.Error()))
```

Comment:
I think this should be continueWithError

Reply:
changed to continueWithError